### ISSUE TITLE
Power Spectrum Mode fails 

### ISSUE THREAD

**Issue by jere19** [pyqtgraph/pyqtgraph#534]
Hello,
I've been using the power spectrum mode of the PlotItem. 
It seems to be broken now in my current installation. It crashes in my own applications and also in pyqtgraph own examples.
I've included the traceback and to me, it seems to be a change in numy API.
Hope this can be solved.

Used packages versions:
- pyqtgraph 0.10.0
- pyqt 5.6.0
- python 3.6.1
- numpy 1.12.1

```
Traceback (most recent call last):
  File "/home/jlaforet/miniconda3/lib/python3.6/site-packages/pyqtgraph/graphicsItems/PlotItem/PlotItem.py", line 891, in updateSpectrumMode
    c.setFftMode(b)
  File "/home/jlaforet/miniconda3/lib/python3.6/site-packages/pyqtgraph/graphicsItems/PlotDataItem.py", line 205, in setFftMode
    self.updateItems()
  File "/home/jlaforet/miniconda3/lib/python3.6/site-packages/pyqtgraph/graphicsItems/PlotDataItem.py", line 483, in updateItems
    x,y = self.getData()
  File "/home/jlaforet/miniconda3/lib/python3.6/site-packages/pyqtgraph/graphicsItems/PlotDataItem.py", line 525, in getData
    x,y = self._fourierTransform(x, y)
  File "/home/jlaforet/miniconda3/lib/python3.6/site-packages/pyqtgraph/graphicsItems/PlotDataItem.py", line 683, in _fourierTransform
    y = abs(f[1:len(f)/2])
TypeError: slice indices must be integers or None or have an __index__ method
```

**Comment by jere19** [comment]
Can be solved by changing line **683** in **pyqtgraph/graphicsItems/PlotDataItem.py** to:
`y = abs(f[1:len(f)//2])`
Using the integer division operator works.

**Comment by campagnola** [comment]
I believe this issue was fixed by #429

**Comment by StSav012** [comment]
The issue is not fixed in the version available via `pip` (0.10.0). It's more than 3 years since the fix is proposed! Unbelievable!

**Comment by j9ac9k** [comment]
We have v0.11rc0 available on pypi you can install, which this fix should be part of.

If you would like the stable release to be on pypi, helping address some of the outstanding show stopper issues will speed up the process, thanks!
